crates/types/src/string.rs

```diff
@@ -22,6 +22,17 @@ pub struct String {
     pub(super) size: usize,
 }
 
+/// A builder that can be used to efficiently build a [`nvim_oxi::String`](String).
+#[repr(C)]
+pub struct StringBuilder {
+    /// Pointer to the allocated memory.
+    pub(super) data: *mut ffi::c_char,
+    /// Current length of the string in bytes, excluding the final null byte.
+    pub(super) len: usize,
+    /// Size of the allocated memory in bytes.
+    pub(super) size: usize,
+}
```

Comment:
Nit: we could replace `data` and `len` with a `String` and rename `size` to `cap`. I think that'd make the purpose of a `StringBuilder` clearer.